core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -272,7 +277,11 @@ public ParametersAction merge(@CheckForNull ParametersAction overrides) {
         return parametersAction;
     }
 
+    @SuppressWarnings("unused")
     private Object readResolve() {
+        if (parameters == null) {
+            parameters = Collections.emptyList();
```

Comment:
Unfortunately there is no way to tweak the param value in XSTream without removing final